spec/factories/rule.rb

```diff
@@ -16,6 +16,16 @@
       profile_id { nil }
     end
 
+    trait :with_group do
+      rule_group { association :v2_rule_group, security_guide: security_guide }
+    end
+
+    trait :with_group_hierarchy do
+      rule_group do
+        association :v2_rule_group, parent_count: (1..5).to_a.sample, security_guide: security_guide
```

Comment:
Can you please extract `parent_count` into a transient variable? We should avoid using constants in factories.